lib/src/layer/tile_layer/tile_coordinates.dart

```diff
@@ -70,3 +47,45 @@ class TileCoordinates extends Point<int> {
     return x ^ y << 24 ^ z << 48;
   }
 }
+
+/// Simplifies coordinates in the context of world replications.
+///
+/// On maps with world replications, different tile coordinates may actually
+/// refer to the same "simplified" tile coordinate - the coordinate that starts
+/// from 0.
+/// For instance, on zoom level 0, all tile coordinates can be simplified to
+/// (0,0), which is the only tile.
+/// On zoom level 1, (0, 1) and (2, 1) can be simplified to (0, 1), as they both
+/// mean the bottom left tile.
+/// And when we're not in the context of world replications, we don't have to
+/// simplify the tile coordinates: we just return the same value.
+class TileCoordinatesSimplifier {
```

Comment:
I'm not 100% sure this is necessary. It adds a little more overhead, and ideally we'd skip it all together if `replicatesWorldLongitude` is `false`. This means the method can be made static, and at that point, is the complexity of another class in this space worth it? Would it be better to just pass a conditional around and have a static function somewhere else (like it was before), or maybe pass an anonymous function around to avoid passing the conditional everywhere?
If we're keeping it, maybe a better name would be "Resolver" or "Replicator". Not sure though.

Reply:
@JaffaKetchup I'd have to double-check that, but that's a matter of OOP (rather my way of coding) against performances (nothing wrong with that) (but in that case not sure the performances would be that much impacted).

I remember that basically I had to create this `TileCoordinatesSimplifier` because:
* the `immutable` clause for `TileImageManager` made it impossible to just set a `bool replicatesWorldLongitude` field in `TileImageManager`
* I wanted the `crs` to set the `bool` value instead of explicitly setting (even "forcing") the same `bool` in the `TileImageManager`, especially given that we need the `camera` for that - and it's available through a `MapCamera.of(context)` that somehow needs a call to `didChangeDependencies` and cannot be set at `initState` time

Reply:
Your arguments sound reasonable but what is there reason to keep `TileImageManager` immutable since it's not really a data class?

Reply:
@josxha I'm not familiar with `@immutable`, but from what I've just read we use only one occurrence of it and there doesn't seem to be added value in that metatag here.
I'll recode without `@immutable`: probably cleaner to read and to maintain.

Reply:
Looking good, thanks!